Implement `instant_print_lines` argument

josherrickson opened the pull request:
Addresses #582; adding functionality to remove any time-delay in printing the first n number of lines.

svenstaro commented on src/lib.rs:
```diff
@@ -16,6 +16,7 @@ use modules::{Module, ALL_MODULES};
 lazy_static::lazy_static! {
     pub static ref CTRLC_PRESSED: AtomicBool = AtomicBool::new(false);
     pub static ref SPEED_FACTOR: Mutex<f32> = Mutex::new(1.0);
+    pub static ref INSTANT_PRINT_LINES: Mutex<u32> = Mutex::new(0);
```
Can't you use `AtomicU32` here instead?

josherrickson replied:
When I switch to `AtomicU32`, the value of `INSTANT_PRINT_LINES.load(Ordering::CstSeq)` in io.rs is always 0 regardless of whether I pass in the flag. Any suggestions of where to start looking to address this?

josherrickson replied:
Nevermind, got it. Will push a fix up shortly.

svenstaro commented on src/io.rs:
```diff
@@ -6,13 +6,24 @@ use wasm_bindgen::prelude::*;
+        time::Duration::new(0, 0)
+    } else {
+        time::Duration::from_millis((1.0 / speed_factor * length as f32) as u64)
+    };
```
Please also handle the `wasm32` case.

svenstaro replied:
Yes, `make web` and then launch a web server on `dist/`. You will have to install `trunk`: `cargo install trunk`. An easy web server you can also install with cargo: `cargo install miniserve`. Then do `miniserve dist/` and then just go there with your browser.

josherrickson replied:
I'm unable to get the web server running, either on my local macbook, or a remote RHL server. Both my fork, and a clone from your local version. It gives 404 for the .wasm and .css files.

svenstaro replied:
Try this: `miniserve dist/ --route-prefix genact --spa --index index.html`

svenstaro commented on src/args.rs:
```diff
@@ -37,6 +43,10 @@ pub struct AppConfig {
     pub speed_factor: f32,
 
+    /// Instantly print this many lines
+    #[clap(short, long = "instant-print-lines", default_value = "0", value_parser = parse_instant_print)]
```
Please also implement this for `wasm32`.

svenstaro reviewed (changes requested):
Great first contribution! A few nitpicks but the overall concept looks sound!

svenstaro commented on src/args.rs:
```diff
@@ -12,6 +12,12 @@ fn parse_speed_factor(s: &str) -> Result<f32, String> {
     Ok(value_as_float)
 }
 
+#[cfg(not(target_arch = "wasm32"))]
+fn parse_instant_print(s: &str) -> Result<u32, String> {
+    let value_as_int = s.parse::<u32>().map_err(|e| e.to_string())?;
+    Ok(value_as_int)
+}
```
Do you actually need this? If you just specify `u32`, it should already parse this correctly I think.

svenstaro commented:
Could you also rebase to get rid of the merge commit into your branch?

josherrickson commented:
I've added wasm32 support. I also rebased.

PS I picked `instant-print-lines` just to avoid `-q`, but I suppose `-i` isn't much better - let me know if you'd prefer I use `quickstart` or something else.

svenstaro commented:
Great work! Not bad for your first Rust contribution. :)
